---
title: "UE Neuroepigenetics M2S3 - Primary analyses of sequencing data"
author: "Stephanie Le Gras$^{1}$ - slegras@igbmc.fr"
date: "October 23-24, 2024"
output:
  html_document:
    fig_caption: yes
    toc: yes
    toc_depth: 5
    toc_float: yes
    number_sections: yes
  word_document:
    toc: yes
    toc_depth: '2'
  pdf_document:
    fig_caption: yes
    keep_tex: yes
    toc: yes
    toc_depth: 2
bibliography: references.bib    
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, root.dir="~/Documents/Formations/2024/UE_neuroEpigenetique")
```

$^1$ GenomEast platform, IGBMC

# - Training set up
## - Datasets analyzed during this training
For this training, we will use two datasets:

  * datasets produced by Achour et al [Pubmed](https://pubmed.ncbi.nlm.nih.gov/25784504/). In this project they analyzed transcriptomics (RNA-seq) and epigenomics (ChIP-seq) data in the striatum of Huntington's disease mice. We will focus on the **RNA-seq data**.

The data are publicly available in GEO under the accession number [GSE59572](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE59572). It contains two subseries:

  * [GSE59570](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE59570): for ChIP-seq data
  * [GSE59571](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE59571): for RNA-seq data

## - Tools used during this training
### - Galaxy
Bioinformatics tools will be run through the french instance of Galaxy, [Galaxy France](https://usegalaxy.fr/) in order to analyzed the data.

### - IGV
The Genome browser IGV will be used to visualize the data in a genomics context.

### - Biojupies
Biojupies will be used to run the differential expression analysis.

# - Prepare your Galaxy environment
Galaxy is a tool that allow users to run bioinformatics tools on a high performance computing cluster through a simple web interface. We are going to use the french instance of Galaxy, Galaxy France.

## - Register to Galaxy France...
... if you don't have an account yet. Go to Galaxy France website: https://usegalaxy.fr/ and register.
![][registergalaxy1]
![][registergalaxy2]
![][registergalaxy3]

## - Log in to Galaxy
Go to Galaxy France website: https://usegalaxy.fr/ and log in with your personal account.

![][logingalaxy]

## - Import a public history that contains data that will be analyzed during this training
Data analyzed during this training are available in a public history: https://usegalaxy.fr/u/stephanie/h/neuro-epigenetics-training-data. Click on the link and import this history or follow the whole process.

### - Browse to the history named "Neuro-epigenetics training (data)"
![][importhistory1]
![][importhistory2]

### - Import the history
![][importhistory3]

### - Create a new working history
![][createnewhistory]

### - Name the new history "Neuro-epigenetics training"
![][changehistoryname]

### - Import raw data (fastq files) from the imported history to the newly created history "Neuro-epigenetics training"
The datasets are in the imported history "Copy of 'Neuro-epigenetics training (data)'".

  * Click on the "History options" button on the top right of your history panel and select "Show Histories Side-by-Side"
![][importfastqfiles]

  * Drag and drop the datasets **R6_1_387_St.chr19.fastq.gz** and **WT_320_St.chr19.fastq.gz** from the imported history to the working one.
![][importfastqfiles2]

# - Analysis of RNA-seq data
Analysis of RNA-seq data will be run with the following steps:

  * [Galaxy] Quality controls
  * [Galaxy] Mapping
  * [Galaxy] Generation of visualization tracks
  * [IGV] Visualization of the data
  * [not done] Generation of per gene counts matrix
  * [Biojupies] Differential expression analysis

![][rnaseqworkflow]

## - Quality controls
Tool: **FastQC**

Website: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/

Citation: Andrews, S. (2010). FastQC:  A Quality Control Tool for High Throughput Sequence Data [Online]. Available online at: http://www.bioinformatics.babraham.ac.uk/projects/fastqc/

Use of the tool: It is used to assess the quality of high throughput sequencing data. The tool takes raw sequencing data (fastq files) or mapping results (BAM, SAM files) and generates a HTML report that gives a quick impression with summary graphs of the quality of the data.

### - Search for the term "fastqc" in the top left search field and click on the tool name "FastQC"
![][searchfastqc]

### - Run Fastqc on the WT sample (WT_320_St.chr19.fastq.gz)
![][runfastqc]

### - Do the same on the R6/1 sample

## - Mapping
Tool: **STAR**

Documentation: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf

Citation: Dobin A, Davis CA, Schlesinger F, Drenkow J, Zaleski C, Jha S, Batut P, Chaisson M, Gingeras TR. STAR: ultrafast universal RNA-seq aligner. Bioinformatics. 2013 Jan 1;29(1):15-21. doi: 10.1093/bioinformatics/bts635. Epub 2012 Oct 25. PMID: 23104886; PMCID: PMC3530905.

Use of the tool: It map RNA-seq data to the reference genome really fast. It uses known transcript junction information to align reads but can also discover new splice junction events.

### - Add the dataset that contains transcript structures (Mus_musculus.NCBIM37.67_UCSConlychr.gtf) to your current history

  * The dataset is in the imported history "Copy of 'Neuro-epigenetics training (data)'".
  * Click on the "History options" button on the top right of your history panel and select "Show Histories Side-by-Side"
![][showhistoryoption]

  * Drag and drop the dataset Mus_musculus.NCBIM37.67_UCSConlychr.gtf from the imported history to the working one.

![][addgtfhistory]

### - Run STAR to map the reads to the genome
![][runstar]

### - Import the two datasets WT_320_St.chr19.bam and R6_1_387_St.chr19.bam to your working history.
As mapping is a long processing step, mapping data are provided in the imported history "Copy of 'Neuro-epigenetics training (data)'".

  * Click on the "History options" button > "Show Histories Side-by-Side"
  * Drag and drop the two datasets from the imported history to the working one.

![][getmappedreads]  

## - Generation of visualization tracks
Tool: **Deeptools bamCoverage**

Documentation: https://deeptools.readthedocs.io/en/develop/content/tools/bamCoverage.html

Citation: Ramírez, Fidel, Devon P. Ryan, Björn Grüning, Vivek Bhardwaj, Fabian Kilpert, Andreas S. Richter, Steffen Heyne, Friederike Dündar, and Thomas Manke. deepTools2: A next Generation Web Server for Deep-Sequencing Data Analysis. Nucleic Acids Research (2016). doi:10.1093/nar/gkw257.

Use of the tool: This is suite of tools is meant to handle next generation sequencing data especially ChIP-seq and RNA-seq data. Some tools can create plots useful to have global views at the data.

### - Use the tool bamCoverage to generate comparable signal tracks from mapping data
![][generatebigwigs]

## - Visualization of the data
### - Download mapping files
Do it for the two files **WT_320_St.chr19.bam** and **R6_1_387_St.chr19.bam**

![][downloadmappingfiles]

### - Download results files (results from bamCoverage)
Do it for the two result datasets.

![][downloadbigwigs]

### - Run the IGV app going to the website https://igv.org/app/, select the assembly: mm9
![][selectassemblyigv]

### - Load the bam files (select both the bam and the bai files) and the bigwig files

In IGV menu:

![][loadfilesigv]

Select the bam, bai and bigwig files.

![][selectfilesigv]

You should get:

![][igvscreenshot]

### - Go to chromosome 19
![][chr19]

### - Select the two bigwig tracks
#### - Set the windowing function to Maximum for the two bigwigs tracks
![][windowmax]

### - Go to Syt12 gene
#### - Set the two bigwig files to the same scale 0-31 at Syt12 gene 
![][samescale1]
![][samescale2]

![][syt12]

## - Generation of per gene counts matrix
### - The matrix of read counts per gene is available in GEO website
It has been downloaded from [GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE59571). It is available in the file https://github.com/slegras/UEneuroepigenetics/blob/main/data/GSE59571_S13113_readCounts.xlsx (follow the link and click on "Download" on the right). We are going to run a differential expression analysis using these data.

## - Differential expression analysis
### - Use the matrix in the tool Biojupies to run a differential expression analysis
Tool: **Biojupies**

Website: https://maayanlab.cloud/biojupies/

Documentation: https://maayanlab.cloud/biojupies/help

Citation: Torre D, Lachmann A, Ma'ayan A. BioJupies: Automated Generation of Interactive Notebooks for RNA-Seq Data Analysis in the Cloud. Cell Syst. 2018 Nov 28;7(5):556-561.e3. doi: 10.1016/j.cels.2018.10.007. Epub 2018 Nov 14. PMID: 30447998; PMCID: PMC6265050.

Use of the tool: BioJupies is a web application that enables RNA-seq data analyses. Through an intuitive interface, users can rapidly generate tailored reports to analyze and visualize their own raw sequencing files, gene expression tables, or fetch data from >9,000 published studies containing >300,000 preprocessed RNA-seq samples.

### - Start an analysis with Biojupies
![][biojupiesetup1]

![][biojupiesetup2]

![][biojupiesetup3]

![][biojupiesetup4]

![][biojupiesetup5]

![][biojupiesetup6]

![][biojupiesetup7]

![][biojupiesetup8]

![][biojupiesetup9]

![][biojupiesetup10]

![][biojupiesetup11]

![][biojupiesetup12]

![][biojupiesetup13]

The created notebook is available here: https://maayanlab.cloud/biojupies/notebook/8Zj5sTemj
or click to run the analysis report.

![][biojupiesok]

### - Download the list of deregulated genes, is Syt12 significantly deregulated?

[registergalaxy1]: images/0-registergalaxy1.png "Register Galaxy 1"
[registergalaxy2]: images/0-registergalaxy2.png "Register Galaxy 2"
[registergalaxy3]: images/0-registergalaxy3.png "Register Galaxy 3"
[logingalaxy]: images/1-logInGalaxy.png "Log In Galaxy"
[importhistory1]: images/2-ImportHistory1.png "Import History 1"
[importhistory2]: images/2-ImportHistory2.png "Import History 2"
[importhistory3]: images/3-ImportHistory3.png "Import History 3"
[createnewhistory]: images/4-CreateNewHistory.png "Create new History"
[changehistoryname]: images/5-ChangeHistoryName.png "Change history name"
[importfastqfiles]: images/6-ImportFastqFiles1.png "import fasta file 1"
[importfastqfiles2]: images/6-ImportFastqFiles2.png "import fasta file 2"
[rnaseqworkflow]: images/7-RNAseqWorkflow.png "RNAseq workflow"
[searchfastqc]: images/8-SearchFastqc.png "Search fastqc"
[runfastqc]: images/9-RunFastqc.png "run fastqc"
[showhistoryoption]: images/10-ShowHistoryOptions.png "Show history options"
[addgtfhistory]: images/11-AddGTFhistory.png "add GTF history"
[runstar]: images/12-RunSTAR.png "run star"
[getmappedreads]: images/13-getmappedReads.png "get mapped reads"
[generatebigwigs]: images/14-generateBigwigs.png "generate bigwigs"
[downloadmappingfiles]: images/15-downloadmappingfiles.png "download mapping files"
[downloadbigwigs]: images/16-downloadbigwigs.png "download bigwig files"
[selectassemblyigv]: images/17-selectAssemblyIGV.png "Select assembly igv"
[loadfilesigv]: images/18-loadfilesIGV.png "load files igv"
[selectfilesigv]: images/19-selectFilesIGV.png "select files igv"
[igvscreenshot]: images/20-igv.png "igv"
[chr19]: images/21-igvChr19.png "chr19"
[samescale1]: images/22-igvsamescale1.png "same scale 1"
[samescale2]: images/22-igvsamescale2.png "same scale 2"
[windowmax]: images/23-igvMaximum.png "windowing function: maximum"
[syt12]: images/24-igvSyt12.png "Syt12"
[biojupiesetup1]: images/25-biojupiesSetup1.png "Start biojupies"
[biojupiesetup2]: images/25-biojupiesSetup2.png "Start biojupies"
[biojupiesetup3]: images/25-biojupiesSetup3.png "Start biojupies"
[biojupiesetup4]: images/25-biojupiesSetup4.png "Start biojupies"
[biojupiesetup5]: images/25-biojupiesSetup5.png "Start biojupies"
[biojupiesetup6]: images/25-biojupiesSetup6.png "Start biojupies"
[biojupiesetup7]: images/25-biojupiesSetup7.png "Start biojupies"
[biojupiesetup8]: images/25-biojupiesSetup8.png "Start biojupies"
[biojupiesetup9]: images/25-biojupiesSetup9.png "Start biojupies"
[biojupiesetup10]: images/25-biojupiesSetup10.png "Start biojupies"
[biojupiesetup11]: images/25-biojupiesSetup11.png "Start biojupies"
[biojupiesetup12]: images/25-biojupiesSetup12.png "Start biojupies"
[biojupiesetup13]: images/25-biojupiesSetup13.png "Start biojupies"
[biojupiesok]: images/26-biojupiesok.png "biojupies ok"
